### PR TITLE
GT: Add platform-related alert pin PLDM event log

### PR DESCRIPTION
**Pull request by Quanta-YangChen**
Summary:
- Change the behavior to load the sensor table by different sources.
- Modify the stage naming way of the GT stage to avoid naming conflicts to the common.
- Add SSD ADC alert pin check and send by different power monitor IC.

Test Plan:
- Build code: Pass

### CODE REVIEW

File: common/service/pldm/pldm_monitor.c [hunk 1]
```diff
@@ -284,11 +288,11 @@ uint8_t pldm_platform_event_message_req(void *mctp_inst, mctp_ext_params ext_par
 	return PLDM_SUCCESS;
 }
 
-static void process_event_message_queue()
+static void process_event_message_queue(struct k_work *work)
 {
```
Comment (garnermic) [inline common/service/pldm/pldm_monitor.c:292]: Please test work for NULL.

File: meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+};
+
+static struct ssd_event_check {
+	struct k_work_delayable work;
+	bool init;
+	uint8_t group;
+	uint8_t flag;
+} ssd_event_check[2];
+
+void ssd_alert_check_ina230(uint8_t idx, uint8_t *flag)
```
Comment (garnermic) [inline meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c:77]: Please test flag for NULL.

File: meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+{
+	uint8_t sensor_num = ssd_power_ic_sensor_table[idx];
+	sensor_cfg *cfg = &sensor_config[sensor_config_index_map[sensor_num]];
```
Comment (garnermic) [inline meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c:80]: Please test cfg for NULL before using.

File: meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+		}
+	}
+	return;
+}
+
+void ssd_alert_check_isl28022(uint8_t idx, uint8_t *flag)
+{
+	uint8_t sensor_num = ssd_power_ic_sensor_table[idx];
+	sensor_cfg *cfg = &sensor_config[sensor_config_index_map[sensor_num]];
+
```
Comment (garnermic) [inline meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c:154]: Please check flag and cfg for NULL before referencing.

File: meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+			LOG_ERR("SSD%d with sensor number(0x%x) post reading failed", idx,
+				cfg->num);
+		}
+	}
+	return;
+}
+
+void ssd_alert_check_handler(struct k_work *work)
+{
+	struct ssd_event_check *data = CONTAINER_OF(work, struct ssd_event_check, work);
```
Comment (garnermic) [inline meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c:236]: Please test work and data for NULL before using.

File: meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+void ssd_alert_check(uint8_t group)
+{
+	if (group > 1) {
+		LOG_ERR("The argument value is invalid");
+		return;
+	}
+
+	struct ssd_event_check *data = ((group == 0) ? &ssd_event_check[0] : &ssd_event_check[1]);
+
+	if (!data->init) {
```
Comment (garnermic) [inline meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c:270]: Please test data for NULL before using.

**Review by garnermic — CHANGES_REQUESTED** [review]
Please resolve the 6 issues as mentioned in my review comments.

**Comment by Quanta-YangChen** [comment]
Hi @garnermic ,

Already update a [commit](https://github.com/facebook/OpenBIC/pull/798/commits/c23ed4e60b500713d5e8cc60f36168fa1ff736cf) modification as suggested. Please review it again, thank you.

**Comment by garnermic** [comment]
Please resolve issues in the commit they are reported as it makes it difficult to verify the fixes and to ensure everything builds correctly in case the follow-on commit doesn't get merged before a build is needed.

**Comment by Quanta-YangChen** [comment]
Hi @garnermic , 

PR #795, #796, #797, and #798 have been successfully built pass on all platforms, please help review them again. Thank you.

**Comment by Quanta-YangChen** [comment]
Close this PR, and will resolve all checks in later PR.

**Pull request closed without merging**